### PR TITLE
General - SQF Improvements

### PR DESCRIPTION
**Pull request by BrettMayson**
**When merged this pull request will:**
- Use `[false, true] select (condition)` when using constants, supposedly 35% faster
- Don't use `typeName` on constants (only in included BI code)
- Use `str ...` instead of `format ["%1", ...]`
- Use `parseNumber ...` instead of `[0, 1] select ...`

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/advanced_ballistics/functions/fnc_initializeTerrainExtension.sqf [hunk 1]
```diff
@@ -53,7 +53,7 @@ INFO_2("Starting Terrain Extension [cells: %1] [world: %2]", _gridCells, worldNa
         private _gridCenter = [_x + 25, _y + 25];
         private _gridHeight = round(getTerrainHeightASL _gridCenter);
         private _gridNumObjects = count (_gridCenter nearObjects ["Building", 50]);
-        private _gridSurfaceIsWater = if (surfaceIsWater _gridCenter) then {1} else {0};
+        private _gridSurfaceIsWater = [0, 1] select (surfaceIsWater _gridCenter);
```
Comment (LinkIsGrim) [inline addons/advanced_ballistics/functions/fnc_initializeTerrainExtension.sqf:56]: parseNumber here instead?
```suggestion
        private _gridSurfaceIsWater = parseNumber (surfaceIsWater _gridCenter);
```

File: addons/fortify/functions/fnc_createObjectMarker.sqf [hunk 2]
```diff
@@ -29,7 +29,7 @@ private _direction = getDir _object;
 
 // Marker name unique to this object
 private _markerNameStr = format [QGVAR(marker_%1), hashValue _object];
-private _channel = if (GVAR(markObjectsOnMap) == 2) then { 0 } else { 1 };
+private _channel = [1, 0] select (GVAR(markObjectsOnMap) == 2);
```
Comment (LinkIsGrim) [inline addons/fortify/functions/fnc_createObjectMarker.sqf:32]:
```suggestion
private _channel = parseNumber (GVAR(markObjectsOnMap) != 2);
```

File: addons/towing/functions/fnc_addRopeToVehicle.sqf [hunk 3]
```diff
@@ -20,8 +20,8 @@ params ["_vehicle"];
 
 if (0 == getNumber (configOf _vehicle >> QEGVAR(cargo,hasCargo))) exitWith {};
 
-private _ropeType = if (
+["ACE_rope6", "ACE_rope12"] select (
```
Comment (Dystopian) [inline addons/towing/functions/fnc_addRopeToVehicle.sqf:23]:
```suggestion
private _ropeType = ["ACE_rope6", "ACE_rope12"] select (
```

**Review by Dystopian — CHANGES_REQUESTED** [review]

File: addons/weather/functions/fnc_getApproximateAirTemp.sqf [hunk 4]
```diff
@@ -18,7 +18,7 @@
 params ["_unit"];
 TRACE_1("params",_unit);
 
-if (isNil (format ["%1", _unit getVariable "ACE_airTemperatureBias"])) then {
+if (isNil (str (_unit getVariable "ACE_airTemperatureBias"))) then {
```
Comment (LinkIsGrim) [inline addons/weather/functions/fnc_getApproximateAirTemp.sqf:21]:
```suggestion
if (isNil {_unit getVariable "ACE_airTemperatureBias"}) then {
```

File: addons/artillerytables/functions/fnc_rangeTableOpen.sqf [hunk 5]
```diff
@@ -65,7 +65,7 @@ GVAR(magModeData) = [];
 {
     _x params ["_xDisplayNameShort", "_xDisplayName", "_xInitSpeed", "_xAirFriction"];
     if (_allSameCharge) then {
-        _ctrlChargeList lbAdd format ["%1", _xDisplayNameShort];
+        _ctrlChargeList lbAdd str _xDisplayNameShort;
```
Comment (PabstMirror) [inline addons/artillerytables/functions/fnc_rangeTableOpen.sqf:68]:
```
format ["%1", "test"]; = "test"
str "test"; = """test"""
```

in this case we can easily do the following, but I'm not sure if that's always true
```suggestion
        _ctrlChargeList lbAdd _xDisplayNameShort;
```

File: addons/common/functions/fnc_stringToColoredText.sqf [hunk 6]
```diff
@@ -18,7 +18,7 @@
 
 params ["_string", "_color"];
 
-_string = format ["%1", _string];
+_string = str _string;
```
Comment (PabstMirror) [inline addons/common/functions/fnc_stringToColoredText.sqf:21]: _string is any, we really need the old way

File: addons/artillerytables/functions/fnc_rangeTableOpen.sqf (hunk 5 above)
Comment (BrettMayson) [reply addons/artillerytables/functions/fnc_rangeTableOpen.sqf:68]: Ah, interesting

**Review by LinkIsGrim — APPROVED** [review]

**Comment by jonpas** [comment]
I don't like that we are changing includes. HEMTT should have an option to ignore warnings in includes.

File: addons/common/functions/fnc_watchVariable.sqf [hunk 7]
```diff
@@ -71,10 +71,10 @@ if (isNull _trackedDisplay) then {
                             (_ctrlGroup controlsGroupCtrl 1) ctrlSetStructuredText parseText format ["%1 (<t color='#00FF00'>%2</t>)", _result, _delta];
                         };
                     } else {
-                        (_ctrlGroup controlsGroupCtrl 1) ctrlSetStructuredText parseText format ["%1", _result];
+                        (_ctrlGroup controlsGroupCtrl 1) ctrlSetStructuredText parseText str _result;
                     };
                 } else {
-                    (_ctrlGroup controlsGroupCtrl 1) ctrlSetStructuredText parseText format ["%1", _result];
+                    (_ctrlGroup controlsGroupCtrl 1) ctrlSetStructuredText parseText str _result;
```
Comment (PabstMirror) [inline addons/common/functions/fnc_watchVariable.sqf:77]:
```suggestion
                    (_ctrlGroup controlsGroupCtrl 1) ctrlSetStructuredText parseText format ["%1", _result];
```

File: addons/recoil/functions/fnc_camshake.sqf [hunk 8]
```diff
@@ -52,8 +52,8 @@ if (isNil "_recoil") then {
     TRACE_3("Caching Recoil config",_weapon,_muzzle,_recoil);
 
     // parse numbers
-    _recoil set [0, call compile format ["%1", _recoil select 0]];
-    _recoil set [1, call compile format ["%1", _recoil select 1]];
+    _recoil set [0, call compile str _recoil select 0];
+    _recoil set [1, call compile str _recoil select 1];
```
Comment (PabstMirror) [inline addons/recoil/functions/fnc_camshake.sqf:56]: > Error position: <linearConversion [0, 1, random 1, _recoi>
 Error Type String, expected Number
File z\ace\addons\recoil\functions\fnc_camshake.sqf..., line 65

```suggestion
    _recoil set [0, call compile format ["%1", _recoil select 0]];
    _recoil set [1, call compile format ["%1", _recoil select 1]];
```

File: addons/xm157/functions/fnc_weaponInfo_draw.sqf [hunk 9]
```diff
@@ -98,7 +98,7 @@ _ctrl ctrlSetText _rangeInfo;
 private _bearingInfo = call {
     private _bearingSetting = GVAR(data) getOrDefault ["bearing_show", 0];
     if ((_bearingSetting == 2) && {_timeSinceLastInput > 2}) exitWith { "" };
-    if ((_bearingSetting == 1)) exitWith { format ["%1", floor (17.777777 * _weaponDir)]; }; // (6400 Mils, not MRAD)
+    if ((_bearingSetting == 1)) exitWith { str floor 17.777777 * _weaponDir; }; // (6400 Mils, not MRAD)
```
Comment (PabstMirror) [inline addons/xm157/functions/fnc_weaponInfo_draw.sqf:101]: Error *: Type String, expected Number,Not a Number
```suggestion
    if ((_bearingSetting == 1)) exitWith { str floor (17.777777 * _weaponDir); }; // (6400 Mils, not MRAD)
```

**Review by PabstMirror — APPROVED** [review]

**Comment by johnb432** [comment]
I'm getting build errors:
```
19:06:29   Error Missing [
19:06:29 File z\ace\addons\hellfire\functions\fnc_getAttackProfileSettings.sqf..., line 37
19:06:29 Error in expression <tClear > 0) then {
1; 
} else {
[
3,
2,
] select (_seekerTargetPos isEqualTo [0,>
19:06:29   Error position: <] select (_seekerTargetPos isEqualTo [0,>
19:06:29   Error Missing [
19:06:29 File z\ace\addons\hellfire\functions\fnc_getAttackProfileSettings.sqf..., line 37
```

There was a comma in the second line - I will get a PR to fix it.

**Pull request merged**